.mergify.yml

```diff
@@ -0,0 +1,101 @@
+pull_request_rules:
+  - name: automatic merge for Dependabot pull request that pass CI
+    conditions:
+      - author=dependabot[bot]
+      - check-success=build-workflow-complete
+    actions:
+      merge:
+        method: merge
+
+  # REVIEW MANAGEMENT
+
+  - name: ask alessandrod to review public API changes
+    conditions:
+      - files=xtask/public-api/aya.txt
+      - -closed
+    actions:
+      comment:
+        message: "Hey @alessandrod, this pull request changes the Aya Public API and requires your review."
+      request_reviews:
+        users:
+          - alessandrod
+      label:
+        add:
+          - api/needs-review
+
+  # LABEL MANAGEMENT
+
+  - name: warn on conflicts
+    conditions:
+      - conflict
+    actions:
+      comment:
+        message: "@{{author}}, this pull request is now in conflict and requires a rebase."
+      label:
+        add:
+          - needs-rebase
+
+  - name: remove conflict label if not needed
+    conditions:
+      - -conflict
+    actions:
+      label:
+        remove:
+          - needs-rebase
+
+  - name: add labels for aya PRs
+    conditions:
+      - files~=^aya/
+      - -closed
+    actions:
+      label:
+        add:
+          - aya
+
+  - name: add labels for aya-bpf PRs
+    conditions:
+      - files~=^(bpf/aya-bpf(-cty|bindings|macros)?/)
```

Comment:
this looks wrong? aya-bpf-macros is at the top level, not in bpf/

Please take a look at my original suggestion.

Reply:
This was your suggestion:

> should this be `(^bpf/|aya-bpf-(bindings|macros)/)`?

That would catch `bpf/aya-log-ebpf` also + aya-bpf-bindings are in the bpf directory, not top-level.
I've now gone with something that should catch `aya-bpf-macros` + only the aya-bpf related dirs in `bpf`.

Reply:
Thanks, my suggestion was incorrect.